src/interrupts/pins_config.rs

```diff
@@ -0,0 +1,323 @@
+/// Functions related to INT1_AG and INT2_AG interrupt pins configuration
+///
+/// TO DO:
+/// -
+///
+use super::*;
+
+/// Accelerometer/gyroscope interrupt pins common settings
+#[derive(Debug)]
+pub struct PinConfig {
+    // --- CTRL_REG8 REGISTER ---
+    /// Interrupt pin active level (default 0: active high)
+    pub active_level: IntActive,
+    /// Interrupt pin push-pull or open-drain configuration (default 0: push-pull)
+    pub pin_mode: IntPin,
+}
+
+impl Default for PinConfig {
+    fn default() -> Self {
+        PinConfig {
+            active_level: IntActive::High,
+            pin_mode: IntPin::PushPull,
+        }
+    }
+}
+
+impl PinConfig {
+    /// Returns values to be written to CTRL_REG8 register
+    fn ctrl_reg8(&self) -> u8 {
```

Comment:
I don't see why. It is only used by another function, there's no need to expose it IMO.

Reply:
Maybe I don't understand how it is used. But I am pretty sure you cannot call this from lib.rs. You will get a Rust Compiler Error E0624. [This link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=f8ad62927f4a6a1d2a2769cd7b10a65d) demonstrates the problem.

Reply:
Oh I see. You can't call it if we move it to lib.rs. It was supposed to be called from pins_config.rs. That's kind of a bummer, though, because if it's public, it will also show up in Rust Analyzer hints, won't it? And it's not supposed to be used directly.

Reply:
Pub(crate) should solve it, I believe.

Reply:
Do we need `ctrl_reg8()`? Where do we use this?

Reply:
It's used by `configure_interrupts_pins` to get the two bit flags out of `PinConfig`. It's not strictly necessary, in fact I noticed that I wasn't using it at all, I was just getting these two values one by one from the struct. But this way it's consistent with other functions.

Reply:
I am leaning toward getting rid of `ctrl_reg8()`. We are not exposing this method anyway. Besides us, nobody knows it even exists.

Reply:
I know, but it's just like `ctrl_reg5_xl()` in `accel.rs`. Which is exposed, but should it be? A user of the crate will/should not call it directly, it should only be used by `begin_accel()`. So it's an internal function to make it more organized for us.

Reply:
We do use it, though, in `configure_interrupts_pin`. It's used to get the configuration, which is just two bits in this case. So we could simply skip it, but this way I find it more coherent with other functions.